
syntax = "proto3";

import "google/protobuf/timestamp.proto";

message LocationMessage {
  int32 id = 1;
  int32 person_id = 2;
  float longitude = 3;
  float latitude = 4;
  google.protobuf.Timestamp creation_time = 5;
}

message Empty {
}

message LocationMessageList {
 repeated LocationMessage locations = 1;
}

message UniqueLocationMessage {
  int32 id = 1;
}

service LocationService {
   rpc Create(LocationMessage) returns (LocationMessage);
   rpc GetLocation(UniqueLocationMessage) returns (LocationMessage);
   rpc Get(Empty) returns (LocationMessageList);
}